pkg/signing/signing.go

```diff
@@ -96,10 +103,130 @@ func printSignatures(entity *openpgp.Entity) error {
 	return nil
 }
 
-func genKeyPair(spath string, ppath string) error {
+func syPgpDirPath() string {
+	return filepath.Join(os.Getenv("HOME"), ".sypgp")
+}
+
+func syPgpSecretPath() string {
+	return filepath.Join(syPgpDirPath(), "pgp-secret")
+}
+
+func syPgpPublicPath() string {
+	return filepath.Join(syPgpDirPath(), "pgp-public")
+}
+
+// Create Singularity PGP home folder, secret and public keyring files
+func syPgpPathsCheck() error {
+	if err := os.MkdirAll(syPgpDirPath(), 0700); err != nil {
+		log.Println("could not create singularity PGP directory")
+		return err
+	}
+
+	fs, err := os.OpenFile(syPgpSecretPath(), os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		log.Println("Could not create private keyring file: ", err)
+		return err
+	}
+	fs.Close()
+
+	fp, err := os.OpenFile(syPgpPublicPath(), os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		log.Println("Could not create public keyring file: ", err)
+		return err
+	}
+	fp.Close()
+
+	return nil
+}
+
+func syPgpLoadPrivKeyring() (openpgp.EntityList, error) {
+	if err := syPgpPathsCheck(); err != nil {
+		return nil, err
+	}
+
+	f, err := os.Open(syPgpSecretPath())
+	if err != nil {
+		log.Println("Error trying to open secret keyring file: ", err)
+		return nil, err
+	}
+	defer f.Close()
+
+	el, err := openpgp.ReadKeyRing(f)
+	if err != nil {
+		log.Println("Error while trying to read secret key ring: ", err)
+		return nil, err
+	}
+
+	return el, nil
+}
+
+func syPgpLoadPubKeyring() (openpgp.EntityList, error) {
+	if err := syPgpPathsCheck(); err != nil {
+		return nil, err
+	}
+
+	f, err := os.Open(syPgpPublicPath())
+	if err != nil {
+		log.Println("Error trying to open public keyring file: ", err)
+		return nil, err
+	}
+	defer f.Close()
+
+	el, err := openpgp.ReadKeyRing(f)
+	if err != nil {
+		log.Println("Error while trying to read public key ring: ", err)
+		return nil, err
+	}
+
+	return el, nil
+}
+
+func printEntity(index int, e *openpgp.Entity) {
+	for _, v := range e.Identities {
+		fmt.Printf("%v) U: %v %v %v\n", index, v.UserId.Name, v.UserId.Comment, v.UserId.Email)
```

Comment:
I think we should avoid using any `fmt.*` functions and stick to `log.*` type functions

Reply:
This is a "*printEntity*" debug function.

Reply:
OK, I'll let it slide this time ;)